.github/workflows/tests.yml

```diff
@@ -214,43 +278,45 @@ jobs:
     # things to test. Running more tests is time consuming due to emulation
     # (factor 2-10x slower).
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        # https://github.com/conda-incubator/setup-miniconda#use-a-default-shell
+        shell: bash -l {0}
     strategy:
       fail-fast: false
       matrix:
-        default-channel: ['defaults', 'conda-forge']
         python-version: ['3.11']
-        platform: ['arm64', 'ppc64le']
-    env:
-      PYTHON: ${{ matrix.python-version }}
+        image: ['continuumio/miniconda3:latest', 'condaforge/miniforge3:latest']
+        platform: [arm64, ppc64le]
+
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout Source
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
-        # Equivalent to locally run:
-        #   `docker run --privileged --rm tonistiigi/binfmt --install all`
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v3
 
-      - name: Python linux-${{ matrix.platform }} on ${{ matrix.default-channel }}, ${{ matrix.python-version }} tests
-        run: >
-          docker run
-          --rm -v ${PWD}:/opt/conda-src
+      - name: Run Tests
+        run: docker run
+          --rm
+          --volume ${PWD}:/opt/conda-src
+          --workdir /opt/conda-src
           --platform linux/${{ matrix.platform }}
-          ghcr.io/conda/conda-ci:main-linux-python${{ matrix.python-version }}${{ matrix.default-channel == 'conda-forge' && '-conda-forge' || '' }}
-          /opt/conda-src/dev/linux/qemu.sh
-
-      - name: Upload test results
-        if: '!cancelled()'
-        uses: actions/upload-artifact@v3
-        with:
-          # name has to be unique, to not overwrite uploads of other matrix runs. sha1 is optional and only to differentiate
-          # when locally dowloading and comparing results of different workflow runs.
-          name: test-results-${{ github.sha }}-linux-${{ matrix.platform }}-qemu-${{ matrix.default-channel }}-${{ matrix.python-version }}
-          path: |
-            .coverage
-            test-report.xml
-          retention-days: 1
+          ${{ matrix.image }}
+          bash -lc
+          ". /opt/conda/etc/profile.d/conda.sh
+          && conda create
+            --name test
+            --quiet --yes
+            --file tests/requirements.txt
+            --file tests/requirements-ci.txt
+            python=${{ matrix.python-version }}
+          && conda activate test
+          && conda info --verbose
+          && conda list --show-channel-urls
+          && pytest tests/test_api.py::test_DepsModifier_contract"
```

Comment:
We could combine this with the benchmark tests instead of this arbitrary old test?

Reply:
we could but why? combining them would results in benchmarks being run on all three archs (`linux-64`, `linux-arm64`, and `linux-ppc64le`)